src/modules/loadPage.ts

```diff
@@ -5,17 +5,21 @@ import { PageRecord } from './Cache';
 export type TransitionOptions = {
 	url: string;
 	customTransition?: string;
+	skipChecks?: boolean;
 };
 
-export const loadPage = function (
-	this: Swup,
-	data: TransitionOptions,
-	popstate: PopStateEvent | null
-) {
-	const { url, customTransition } = data;
+export const loadPage = function (this: Swup, data: TransitionOptions, popstate?: PopStateEvent) {
+	const { url, customTransition, skipChecks = false } = data;
+
+	// Check if the visit should be ignored
+	if (!skipChecks && this.shouldIgnoreVisit(url)) {
+		window.location.href = url;
+		return;
+	}
```

Comment:
I really struggle with the idea of doing the check conditionally here the second time in the process. Since the problem is `swup.loadPage` provided on Swup to the consumer, how about we introduce a wrapper around this method with this name. Basically, it wouldn't be provided on the instance directly in this form. 
The wrapper function would be the actual accessible `loadPage` and consist of the checks, later calling this method, just like the click and popstate handlers do?

Reply:
Yeah, I've been thinking the same thing. How could we call the internal API method that wouldn't do the check -- performPageLoad? executePageLoad?

Reply:
That would also let us get rid of the internal popsate argument on the external method.

Reply:
Both will do, it doesn't matter much since it can be always changed. Maybe `performNavigation` or `startTransition`?